System/Metrics/Distribution.hsc

```diff
@@ -132,34 +132,123 @@ new = (Distribution . fromList numStripes) `fmap`
+    case mean +#### ((int2Double## n) *#### delta /#### (int2Double## count')) of { mean' ->
+    case sumSqDelta +#### (delta *#### (val -#### mean') *#### (int2Double## n)) of { sumSqDelta' ->
+    case writeIntArray## mba countPos' count' s8 of { s9 ->
+    case writeDoubleArray## mba meanPos' mean' s9 of { s10 ->
+    case writeDoubleArray## mba sumSqDeltaPos' sumSqDelta' s10 of { s11 ->
+    case writeDoubleArray## mba sumPos' (dSum +#### val) s11 of { s12 ->
+    case (case val <#### dMin of { 0## -> dMin; _ -> val }) of { dMin' ->
+    case (case val >#### dMax of { 0## -> dMax; _ -> val }) of { dMax' ->
+    case writeDoubleArray## mba minPos' dMin' s12 of { s13 ->
+    case writeDoubleArray## mba maxPos' dMax' s13 of { s14 ->
+    case spinUnlock mba s14 of { s15 ->
+    (## s15, () ##) }}}}}}}}}}}}}}}}}}}}}}}}}}}}
+
+combine :: Distrib -> Distrib -> IO ()
+combine (Distrib bMBA) (Distrib aMBA) = IO $ \s ->
+    case spinLock bMBA s of { s1 ->
+    case countPos of { (I## countPos') ->
+    case readIntArray## aMBA countPos' s1 of { (## s2, aCount ##) ->
+    case readIntArray## bMBA countPos' s2 of { (## s3, bCount ##) ->
+    case aCount +## bCount of { count' ->
+    case meanPos of { (I## meanPos' ) ->
+    case readDoubleArray## aMBA meanPos' s3 of { (## s4, aMean ##) ->
+    case readDoubleArray## bMBA meanPos' s4 of { (## s5, bMean ##) ->
+    case bMean -#### aMean of { delta ->
+    case (   (((int2Double## aCount) *#### aMean) +#### ((int2Double## bCount) *#### bMean))
+         /#### (int2Double## count')
+         ) of { mean' ->
+    case sumSqDeltaPos of { (I## sumSqDeltaPos') ->
+    case readDoubleArray## aMBA sumSqDeltaPos' s5 of { (## s6, aSumSqDelta ##) ->
+    case readDoubleArray## bMBA sumSqDeltaPos' s6 of { (## s7, bSumSqDelta ##) ->
+    case (   aSumSqDelta 
+         +#### bSumSqDelta
+         +#### (   delta
+               *#### delta
+               *#### (   (int2Double## aCount) *#### (int2Double## bCount)
+                     /#### (int2Double## count')
+                     )
+               )
+         ) of { sumSqDelta' ->
+    case writeIntArray## aMBA countPos' count' s7 of { s8 ->
+    case (case count' ==## 0## of { 0## -> mean'; _ -> 0.0#### }) of { writeMean ->
+    case writeDoubleArray## aMBA meanPos' writeMean s8 of { s9 ->
+    case writeDoubleArray## aMBA sumSqDeltaPos' sumSqDelta' s9 of { s10 ->
+    case sumPos of { (I## sumPos') ->
+    case readDoubleArray## aMBA sumPos' s10 of { (## s11, aSum ##) ->
+    case readDoubleArray## bMBA sumPos' s11 of { (## s12, bSum ##) ->
+    case writeDoubleArray## aMBA sumPos' (aSum +#### bSum) s12 of { s13 ->
+    case minPos of { (I## minPos') ->
+    case readDoubleArray## bMBA minPos' s13 of { (## s14, bMin ##) ->
+    case writeDoubleArray## aMBA minPos' bMin s14 of { s15 ->
+    case maxPos of { (I## maxPos') ->
+    case readDoubleArray## bMBA maxPos' s15 of { (## s16, bMax ##) ->
+    case writeDoubleArray## aMBA maxPos' bMax s16 of { s17 ->
```

Comment:
https://github.com/haskell-github-trust/ekg-core/pull/42/files?diff=unified&w=0#diff-39ca52d320496f56a11be22daa330e8a324089b72e6c2124240e7326b5907aabL41-L42

You should include this comment to justify the odd swap.

Reply:
Sorry but maybe the link is broken? It just takes me to the top of this diff, so I don't know exactly what you're talking about here.

Reply:
Ah, so in the c code you take this from there's a comment about how doing the value swapping is a little bit of a hack and optimisation, with a further link to the commit where they worked it out. Keeping that explanation would be useful.